2002-12-26  Anton Kachalov <mouse@altlinux.org>

	* apt-pkg/packagemanager.cc (pkgPackageManager::CheckRConflicts):
	Ignore versionless reverse dependencies.

--- apt-0.5.4cnc9/apt-pkg/packagemanager.cc.orig	2002-12-26 21:29:50 +0300
+++ apt-0.5.4cnc9/apt-pkg/packagemanager.cc	2002-12-26 21:30:48 +0300
@@ -216,7 +216,7 @@ bool pkgPackageManager::CheckRConflicts(
       if (D.ParentPkg() == Pkg || D.ParentVer() != D.ParentPkg().CurrentVer())
 	 continue;
       
-      if (Cache.VS().CheckDep(Ver,D) == false) // CNC:2002-07-10
+      if (D.TargetVer() == 0 || Cache.VS().CheckDep(Ver,D) == false) // CNC:2002-07-10
 	 continue;
 
       if (EarlyRemove(D.ParentPkg()) == false)
